### ISSUE TITLE
Feature request to restrict mouse movement

### ISSUE THREAD

**Issue by nhrdl** [JoonasVali/NaturalMouseMotion#1]
Thank you for the good work you have done. I really like the library.

I would like to limit mouse movement in certain area of the screen only. Its not clear from reading the code if it can be done or not.

My use case is kind of plotting points on a canvas. I just want to send events to my canvas area and ignore other parts of screen.

**Comment by JoonasVali** [comment]
Hi, thank you. I think this idea is certainly worth looking into and I will try to delve deeper into it soon.
Meanwhile here's a workaround for you that might actually do exactly what you need:

(This code works in NaturalMouseMotion 1.0.3, but from next release 1.1.0, there are some API changes where this won't compile)
```
import com.github.joonasvali.naturalmouse.api.MouseMotionFactory;
import com.github.joonasvali.naturalmouse.support.DefaultSystemCalls;
import com.github.joonasvali.naturalmouse.support.MouseInfoAccessor;

import java.awt.*;
import java.awt.event.InputEvent;

public class Main {

  private static final int CANVAS_WIDTH = 350;
  private static final int CANVAS_HEIGHT = 300;
  private static final int CANVAS_X = 200;
  private static final int CANVAS_Y = 200;

  public static void main(String[] args) throws InterruptedException {
    Thread.sleep(3000L);
    MouseMotionFactory factory = new MouseMotionFactory();
    factory.setMouseMovementBaseMs(500);

    // Override screen size to match your canvas.
    factory.setSystemCalls(new DefaultSystemCalls() {
      Dimension dimension = new Dimension(CANVAS_WIDTH, CANVAS_HEIGHT);
      @Override
      public Dimension getScreenSize() {
        return dimension;
      }
    });

    // If your canvas is not at 0,0 at the screen, you want to make the Robot to have an offset
    factory.setRobot(createOffsetRobot(CANVAS_X, CANVAS_Y));
    // For the same reason you need to counter your offset when fetching mouse pointer info,
    // so it would think (CANVAS_X, CANVAS_Y) is (0, 0).
    factory.setMouseInfo(createOffsetMouseInfo(CANVAS_X, CANVAS_Y));


    // If you open 'paint' and activate the pencil tool, it'll draw a box around where the canvas should be.
    factory.move(0, 0);
    factory.getRobot().mousePress(InputEvent.BUTTON1_MASK);
    factory.move(0, CANVAS_HEIGHT);
    factory.move(CANVAS_WIDTH, CANVAS_HEIGHT);
    factory.move(CANVAS_WIDTH, 0);
    factory.move(0, 0);
    factory.getRobot().mouseRelease(InputEvent.BUTTON1_MASK);
    // End of drawing.

    Thread.sleep(1000);
    factory.move(0, 0); // Moves mouse to 200, 200 on screen
    // Tries to move mouse to 700, 700 on screen, but the actual maximum
    // mouse position is (200 + 350, 200 + 300) = (550, 500), so it limits the movement there.
    factory.move(500, 500);


  }

  private static MouseInfoAccessor createOffsetMouseInfo(int xOffset, int yOffset) {
    return new MouseInfoAccessor() {
      Point p = new Point();
      @Override
      public Point getMousePosition() {
        Point realPointer = MouseInfo.getPointerInfo().getLocation();
        p.setLocation(realPointer.x - xOffset, realPointer.y - yOffset);
        return p;
      }
    };
  }

  private static Robot createOffsetRobot(int xOffset, int yOffset) {
    try {
      return new Robot() {
        @Override
        public synchronized void mouseMove(int x, int y) {
          super.mouseMove(x + xOffset, y + yOffset);
        }
      };
    } catch (AWTException e) {
      throw new RuntimeException(e);
    }
  }
}
```

**Comment by nhrdl** [comment]
Sounds similar to approach I took, basically coordinate translation. I'm setting Dimension to interested area and then doing the actual coordinate mapping on my own. 

Would be nice to have have core framework support. Basically, I want to to be able to say restrict mouse movements to this area.

**Comment by JoonasVali** [comment]
Yes, I'm trying to come up with a simple solution for this :)

**Comment by JoonasVali** [comment]
This feature is now implemented since Natural Mouse Motion 2.0.0. https://github.com/JoonasVali/NaturalMouseMotion#translating-coordinates-from-real-screen-to-virtual-screen
